subsys/bluetooth/shell/bt.c

```diff
@@ -537,17 +540,55 @@ static void bt_ready(int err)
 static int cmd_init(const struct shell *shell, size_t argc, char *argv[])
 {
 	int err;
+	bool no_ready_cb = false;
 
 	ctx_shell = shell;
 
-	err = bt_enable(bt_ready);
-	if (err) {
-		shell_error(shell, "Bluetooth init failed (err %d)", err);
+	for (size_t argn = 1; argn < argc; argn++) {
+		const char *arg = argv[argn];
+
+		if (!strcmp(arg, "no-settings-load")) {
+			no_settings_load = true;
+		} else if (!strcmp(arg, "no-ready-cb")) {
```

Comment:
"sync" would (at least to me) be a bit more clearer than "no-ready-cb", since really what the callback impacts is whether the initialisation happens synchronously or asynchronously.

Reply:
Changed to "sync" instead